src/nvim/option.c

```diff
@@ -463,12 +457,31 @@ static OptVal get_option_default(const OptIndex opt_idx, int opt_flags)
   }
 }
 
+/// Allocate the default values for all options by copying them from the stack.
+/// This ensures that we don't need to always check if the option default is allocated or not.
+static void alloc_options_default(void)
+{
+  for (OptIndex opt_idx = 0; opt_idx < kOptIndexCount; opt_idx++) {
+    options[opt_idx].def_val = optval_copy(options[opt_idx].def_val);
+  }
+}
+
+/// Change the default value for an option.
+///
+/// @param  opt_idx  Option index in options[] table.
+/// @param  value    New default value. Must be allocated.
+static void set_option_default(const OptIndex opt_idx, OptVal value)
+{
+  optval_free(options[opt_idx].def_val);
+  options[opt_idx].def_val = value;
+}
+
 /// Set an option to its default value.
 /// This does not take care of side effects!
 ///
 /// @param  opt_idx    Option index in options[] table.
 /// @param  opt_flags  Option flags.
-static void set_option_default(const OptIndex opt_idx, int opt_flags)
+static void set_option_to_default(const OptIndex opt_idx, int opt_flags)
```

Comment:
may be confusing for little gain to redefine (line 473 above) the meaning of `set_option_default` (compared to Vim). 

instead we could keep that name here, and use a name like `def_option_default` ("define").

Reply:
On second thought, `modify_option_default` seems like a better name